### PR TITLE
Support for Lentainform

### PR DESCRIPTION
**Pull request by DenyHF**

### CODE REVIEW

File: 3p/integration.js [hunk 1]
```diff
@@ -266,7 +266,7 @@ import {zen} from '../ads/zen';
 import {zergnet} from '../ads/zergnet';
 import {zucks} from '../ads/zucks';
 import {speakol} from '../ads/speakol';
-
+import {lentainform} from '../ads/lentainform';
```
Comment (torch2424) [inline 3p/integration.js]: Can we put this (and the one above, my apologies), in alphabetical order? 😄

File: 3p/integration.js [hunk 2]
```diff
@@ -513,6 +513,7 @@ register('zen', zen);
 register('zergnet', zergnet);
 register('zucks', zucks);
 register('speakol', speakol);
+register('lentainform', lentainform);
```
Comment (torch2424) [inline 3p/integration.js]: Can we put this (and the one above, my apologies), in alphabetical order? 😄

File: ads/_config.js [hunk 3]
```diff
@@ -1127,4 +1127,8 @@ export const adConfig = {
     renderStartImplemented: true,
   },
 
+  'lentainform': {
```
Comment (torch2424) [inline ads/_config.js]: Can we put this in alphabetical order? 😄

File: ads/lentainform.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline ads/lentainform.js]: Incorrect copyright year 😄

File: ads/lentainform.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<!---
+Copyright 2015 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline ads/lentainform.md]: Incorrect copyright year 😄

File: ads/lentainform.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+       data-widget="12345"
+       data-container="demo-container">
+  </amp-embed>
+```
+
+## Configuration
+
+For details on the configuration semantics, please contact the ad network or refer to their documentation. 
+
+Supported parameters:
```
Comment (torch2424) [inline ads/lentainform.md]: Can we state which parameters are optional, and which are required? 😄

File: examples/ads.amp.html [hunk 7]
```diff
@@ -249,6 +249,7 @@
         <option>zen</option>
         <option>zergnet</option>
         <option>zucks</option>
+        <option>lentainform</option>
```
Comment (torch2424) [inline examples/ads.amp.html]: Can we put this in alphabetical order? 😄

File: examples/ads.amp.html [hunk 8]
```diff
@@ -2084,5 +2085,13 @@ <h2>Zucks</h2>
           data-zoe-multi-ad="true">
   </amp-ad>
 
+  <h2>Lentainform</h2>
```
Comment (torch2424) [inline examples/ads.amp.html]: Can we put this in alphabetical order? 😄

File: extensions/amp-ad/amp-ad.md [hunk 9]
```diff
@@ -387,6 +387,7 @@ See [amp-ad rules](https://github.com/ampproject/amphtml/blob/master/extensions/
 - [Yieldpro](../../ads/yieldpro.md)
 - [Zedo](../../ads/zedo.md)
 - [Zucks](../../ads/zucks.md)
+- [Lentainform](../../ads/lentainform.md)
```
Comment (torch2424) [inline extensions/amp-ad/amp-ad.md]: Can we put this in alphabetical order? 😄

**Comment by torch2424** [comment]
@DenyHF Thank you very much for making the changes!

So everything looks good in the code, as well as the example:

Screen Shot 2019-05-06 at 1 11 54 PM

However, when I added the following log statement:

Screen Shot 2019-05-06 at 1 04 54 PM

Seems like the `renderStart_` in `amp-ad-xorigin-iframe-handler.js` never gets called even though you have `renderStartImplemented: true`. Is this done on the server side? Or is it something on my end?

Thank you!

**Comment by DenyHF** [comment]
@torch2424 
Hello, this problem is on our side.
I deleted this parameter from the config

File: ads/_config.js [hunk 10]
```diff
@@ -555,6 +555,8 @@ export const adConfig = {
 
   'kuadio': {},
 
+  'lentainform': {},
```
Comment (torch2424) [inline ads/_config.js]: So I noticed here you removed implementing `renderStartImplemented`. If possible, could you please implement this? It helps us serve ads as fast as possible to our users, and gives us some big performance gains.

If it's something you plan on supporting, and it's a server-side thing. I'm okay with merging this, and you can fix on your server 😄 

Let me know what you think! 👍

Comment (DenyHF) [reply ads/_config.js]: Hello, at the moment we do not support this method.
We will implement it soon.
revert back to` renderStartImplemented=true` ?

Comment (torch2424) [reply ads/_config.js]: By @lannka suggestion, let's set this to `renderStartImplemented: false`, and then we can merge this in. Once you implement on the server, we can then open a new PR for `renderStartImplemented: true` 😄 👍

**Comment by torch2424** [comment]
@DenyHF Thank you very much for making these changes!

Merging! 👍

**Pull request merged**